cmd/rpcdaemon/health/check_txpool.go

```diff
@@ -0,0 +1,83 @@
+package health
+
+import (
+	"context"
+	"fmt"
+	"github.com/ledgerwatch/erigon-lib/common/hexutil"
+	"github.com/ledgerwatch/erigon/rpc"
+	"github.com/ledgerwatch/erigon/turbo/jsonrpc"
+)
+
+func checkTxPool(ctx context.Context, txApi TxPoolAPI, ethApi EthAPI) error {
+	if txApi == nil {
+		return fmt.Errorf("no connection to the Erigon server or `tx_pool` namespace isn't enabled")
+	}
+
+	if ethApi == nil {
+		return fmt.Errorf("no connection to the Erigon server or `eth` namespace isn't enabled")
+	}
+
+	var err error
+
+	data, err := txApi.Content(ctx)
+	if err != nil {
+		return err
+	}
+
+	contentMap, ok := data.(map[string]map[string]map[string]*jsonrpc.RPCTransaction)
+	if !ok {
+		return fmt.Errorf("unexpected response type for tx_pool: %T", data)
+	}
+
+	pendingData, ok := contentMap["pending"]
+	if !ok {
+		return nil
+	}
+
+	if pendingData != nil && len(pendingData) > 0 {
+		// pending pool has transactions lets check last block if it was 0 tx
+		var latestBlockData map[string]interface{}
+		fullTx := false
+		latestBlockData, err = ethApi.GetBlockByNumber(ctx, -1, &fullTx)
+		if err != nil {
+			return err
+		}
+
+		var latestBlockNo *hexutil.Big
+		latestBlockNo, ok = latestBlockData["number"].(*hexutil.Big)
+		if !ok {
+			return nil
+		}
+
+		// check 5 blocks back from latest to see if any of them has transactions
+		blockTransactions := make([][]interface{}, 0)
+		for i := 0; i < 5; i++ {
+			blockToCheck := latestBlockNo.ToInt().Int64() - int64(i)
+			var prevBlockData map[string]interface{}
+			prevBlockData, err = ethApi.GetBlockByNumber(ctx, rpc.BlockNumber(blockToCheck), &fullTx)
+			if err != nil {
+				return err
+			}
+			var transactions []interface{}
+			transactions, ok = prevBlockData["transactions"].([]interface{})
+			if !ok {
+				return nil
+			}
+			blockTransactions = append(blockTransactions, transactions)
```

Comment:
here we could save some time / resources by just setting `foundTransactions` to be true as soon as we see a block with some and break the loop.  You might only need to iterate once or twice instead of checking all 5 blocks every time.